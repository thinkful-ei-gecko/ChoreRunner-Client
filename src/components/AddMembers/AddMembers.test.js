import React from 'react';
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom'
import ReactDOM from 'react-dom'
import toJson from 'enzyme-to-json'
import AddMembers from './AddMembers'
import renderer from 'react-test-renderer';
import enzyme from 'enzyme'

describe('AddMembers component', () => {
  
  it('renders an .AddMembers by default', () => {
    const wrapper = shallow(<AddMembers />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <AddMembers />
      </BrowserRouter>
    , div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders UI as expected', () => {
    const tree = renderer
      .create(<AddMembers />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })

  it('should trigger onChange event', () => {
    let changeText = 'text';
    const wrapper = enzyme.shallow(<AddMembers onChange={(value) => changeText = value} />)
    wrapper.find('input[name="text"]').simulate('change', { target: {value: 'text'}});
    expect(changeText).toEqual('text')
  })
})