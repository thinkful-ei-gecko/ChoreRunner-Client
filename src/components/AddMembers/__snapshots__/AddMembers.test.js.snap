// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMembers component renders UI as expected 1`] = `
<div
  className="add-member container"
>
  <p>
    ADD HOUSEHOLD MEMBERS:
  </p>
  <form
    className="add-household-form"
    id="add-household-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="member-name"
    >
      Name
    </label>
    <input
      id="member-name"
      name="name"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <div
      className="valid-error"
    >
      
    </div>
    <label
      htmlFor="household"
    >
      Household
    </label>
    <select
      className="select-css"
      defaultValue="Select household"
      id="assignee"
      name="household_id"
      onChange={[Function]}
      required={true}
      type="text"
    >
      <option
        disabled={true}
      >
        Select household
      </option>
    </select>
    <label
      htmlFor="child-username"
    >
      Child username
    </label>
    <input
      id="child-username"
      name="username"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      htmlFor="child-password"
    >
      Child password
    </label>
    <input
      id="child-password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <button
      className="submitHH"
      type="submit"
    >
      add
    </button>
  </form>
  <p />
</div>
`;
